File: Clases2025-26/Clase12_PersistenciaJava/app/src/main/java/com/example/clase12_persistenciajava/Persona.java
package com.example.clase12_persistenciajava;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "persona")
public class Persona
{
        @PrimaryKey(autoGenerate = true) int id;
        //@ColumnInfo(name ="id") int id,
        @ColumnInfo(name="nombre") String nombre;

        @ColumnInfo(name="edad") int edad;

        @ColumnInfo(name="direccion") String direccion;


    public Persona(int id, String nombre, int edad, String direccion) {
        this.id = id;
        this.nombre = nombre;
        this.edad = edad;
        this.direccion = direccion;
    }


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public int getEdad() {
        return edad;
    }

    public void setEdad(int edad) {
        this.edad = edad;
    }

    public String getDireccion() {
        return direccion;
    }

    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }

    @Override
    public String toString() {
        return "Persona{" +
                "id=" + id +
                ", nombre='" + nombre + '\'' +
                ", edad=" + edad +
                ", direccion='" + direccion + '\'' +
                '}';
    }
}

File: Clases2023-24/Clase31_AccesoBaseDatos_Coroutines/app/src/main/java/com/example/clase31_accesobasedatos_coroutines/Persona.kt
package com.example.clase31_accesobasedatos_coroutines

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey


@Entity(tableName = "persona")
data class Persona(
                    @PrimaryKey(autoGenerate = true) var id: Int,
                    @ColumnInfo var nombre :String,
                   @ColumnInfo var edad : Int,
                   @ColumnInfo var direccion:String) {
    constructor(nombre :String, edad : Int, direccion:String) : this(0,nombre,edad,direccion)
    override fun toString(): String {
        return "Persona(nombre='$nombre', edad=$edad, direccion='$direccion', id=$id)"
    }
}
